### DISCUSSION TITLE
No ExtData: Faster than I expected

### DISCUSSION THREAD

**Discussion by mathomp4** [GEOS-ESM/MAPL#1050] [General]
Dan Duffy has been trying to run GEOS on AWS and was having a weird crash with ESMF_RegridStore during ExtData or something. Until @bena-nasa or myself can get on there and see what's happening, my advice to him was to run without ExtData. In a way, since he's just benchmarking the *processor* more than the disk system, this isn't exactly a bad idea.

But I then had a thought: what is the effect of not running with ExtData. So, I got on discover and ran some 1-day C360 runs. No history, no checkpointing, and one with ExtData, one without. I did three of each and I got:

With ExtData:
```
           MAPL: INFO: Model Throughput:      122.599 days per day
           MAPL: INFO: Model Throughput:      119.870 days per day
           MAPL: INFO: Model Throughput:      123.487 days per day
```

Without ExtData:
```
           MAPL: INFO: Model Throughput:      131.252 days per day
           MAPL: INFO: Model Throughput:      131.680 days per day
           MAPL: INFO: Model Throughput:      129.332 days per day
```

Now, in the timers for the ExtData Run phase:
```
--Run                                                     1   671.594  95.27     0.336   0.05
----EXTDATA                                             192     7.246   1.03     7.246   1.03
```
vs:
```
--Run                                                     1   624.606  94.85     0.332   0.05
----EXTDATA                                             192     0.003   0.00     0.003   0.00
```
So, obviously not running ExtData speeds up the model more than just ExtData itself. I just thought I'd start a discussion and see if @bena-nasa, @atrayano, or @tclune had thoughts. 

I did talk about this yesterday with @bena-nasa (with an earlier run) and he had thoughts that maybe not having any "real" imports mean some calculations in Chem might be skipped and that does seem plausible as GOCART does get cheaper:
```
----------CHEMISTRY                                     576   133.170  18.89     0.086   0.01
------------CHEMENV                                     576     0.713   0.10     0.713   0.10
------------PCHEM                                       384     7.361   1.04     7.361   1.04
------------GOCART                                      576   121.429  17.22   121.429  17.22
------------TR                                          384     3.582   0.51     3.582   0.51
```
to:
```
----------CHEMISTRY                                     576    98.658  14.98     0.084   0.01
------------CHEMENV                                     576     0.691   0.10     0.691   0.10
------------PCHEM                                       384     7.149   1.09     7.149   1.09
------------GOCART                                      576    86.914  13.20    86.914  13.20
------------TR                                          384     3.821   0.58     3.821   0.58
```

I just thought I'd add this here as a place where, if @bena-nasa or someone can track down why, we could have an explanation in case we are asked about it.

**Comment by tclune** [comment]
The problem with the theory is that there is now way for components such as GOCART to "know" if their imports are coming from ExtData or simply set to 0.    My only other thought is that there is variability in the other timers and it just looked like it was dominated by ExtData.   But that seems unlikely as well.

**Comment by mathomp4** [comment]
I suppose if the calculations were protected by an "if not all zero"??? Though I guess I don't know what happens with no ExtData. I can imagine the first step is zero, but after that? I suppose if there are no emissions coming from ExtData, those emissions will always be zero...
